app/controllers/journeys/income/TradingAllowanceAmountController.scala

```diff
@@ -42,38 +42,38 @@ class TradingAllowanceAmountController @Inject() (override val messagesApi: Mess
     extends FrontendBaseController
     with I18nSupport {
 
-  def onPageLoad(taxYear: Int, mode: Mode): Action[AnyContent] = (identify andThen getData) { // TODO add requireData SASS-5841
+  def onSubmit(taxYear: Int, businessId: String, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) async {
     implicit request =>
       val turnoverAmount: BigDecimal = {
-        val turnover: BigDecimal = request.userAnswers.getOrElse(UserAnswers(request.userId)).get(TurnoverIncomeAmountPage).getOrElse(1000.00)
-        if (turnover > 1000.00) 1000.00 else turnover
-      }
-      val preparedForm = request.userAnswers.getOrElse(UserAnswers(request.userId)).get(TradingAllowanceAmountPage) match {
-        case None        => formProvider(authUserType(request.user.isAgent), turnoverAmount)
-        case Some(value) => formProvider(authUserType(request.user.isAgent), turnoverAmount).fill(value)
-      }
-
-      Ok(view(preparedForm, mode, authUserType(request.user.isAgent), taxYear))
-  }
-
-  def onSubmit(taxYear: Int, mode: Mode): Action[AnyContent] = (identify andThen getData) async { // TODO add requireData SASS-5841
-    implicit request =>
-      val turnoverAmount: BigDecimal = {
-        val turnover: BigDecimal = request.userAnswers.getOrElse(UserAnswers(request.userId)).get(TurnoverIncomeAmountPage).getOrElse(1000.00)
+        val turnover: BigDecimal = request.userAnswers.get(TurnoverIncomeAmountPage, Some(businessId)).getOrElse(1000.00)
```

Comment:
It would be nice to keep controllers logic-free. 

- Any ifs and magic numbers ideally could live in models (if we don't return Futures) or in service layer (if we do).
- Also, can you extract 1000 to const with a name so it's clearer what it does

Reply:
will make this change